mergin/client_pull.py

```diff
@@ -103,7 +104,25 @@ def _cleanup_failed_download(directory, mergin_project=None):
     if mergin_project is not None:
         mergin_project.remove_logging_handler()
 
+    # keep log file as it might contain useful debug info
+    log_file = os.path.join(directory, ".mergin", "client-log.txt")
+    dest_file = None
+
+    if os.path.exists(log_file):
+        dest_file = os.path.join(tempfile.gettempdir(), os.path.split(log_file)[1])
+        head, tail = os.path.split(os.path.normpath(dest_file))
+        ext = "".join(Path(tail).suffixes)
+        file_name = tail.replace(ext, "")
+
+        i = 0
+        while os.path.exists(dest_file):
+            i += 1
+            dest_file = os.path.join(head, file_name) + f"_{i}{ext}"
+
+        shutil.copyfile(log_file, dest_file)
+
     shutil.rmtree(directory)
+    return dest_file
```

Comment:
Worth adding to the docstring that it returns log filename or None if log does not exist...